cdap-support-bundle/src/main/java/io/cdap/cdap/support/services/SupportBundleGenerator.java

```diff
@@ -197,25 +325,102 @@ private File getLatestFolder(File baseDirectory) {
     }
-    return supportBundleStatus;
+    return bundleStatus;
   }
 
+  /**
+   * valid if the namespace exists or not
+   */
   private NamespaceId validNamespace(NamespaceId namespace) throws Exception {
     if (!namespaceQueryClient.exists(namespace)) {
       throw new NamespaceNotFoundException(namespace);
     }
 
     return namespace;
   }
+
+  /**
+   * Collect single support bundle task status
+   */
+  private SupportBundleTaskReportStatus collectSupportBundleTaskStatus(
+    Set<SupportBundleTaskStatus> supportBundleTaskStatusSet) {
+    CollectionState systemLogTaskStatus = CollectionState.INVALID;
+    CollectionState pipelineInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeLogTaskStatus = CollectionState.INVALID;
+    CollectionState vmInfoTaskStatus = CollectionState.INVALID;
+    // Expand all supportBundleTaskStatusSet and map each individual taskType and subTaskType with corresponding status
+    // SupportBundleRuntimeInfoTask & SupportBundlePipelineRunLogTask are two subtasks of SupportBundlePipelineInfoTask
+    Map<SupportBundleTaskType, CollectionState> taskStatusMap = new HashMap<>();
+    for (SupportBundleTaskStatus supportBundleTaskStatus : supportBundleTaskStatusSet) {
+      SupportBundleTaskType currentTaskType = SupportBundleTaskType.valueOf(supportBundleTaskStatus.getType());
+      taskStatusMap.put(currentTaskType, supportBundleTaskStatus.getStatus());
+      if (supportBundleTaskStatus.getSubTasks().size() > 0) {
+        for (SupportBundleTaskStatus subTaskSupportBundleTaskStatus : supportBundleTaskStatus.getSubTasks()) {
+          SupportBundleTaskType currentSubTaskType =
+            SupportBundleTaskType.valueOf(subTaskSupportBundleTaskStatus.getType());
+          taskStatusMap.put(currentSubTaskType, subTaskSupportBundleTaskStatus.getStatus());
+        }
+      }
+    }
+
+    for (SupportBundleTaskType taskType : taskStatusMap.keySet()) {
+      switch (taskType) {
+        case SupportBundleSystemLogTask:
+          systemLogTaskStatus = taskStatusMap.get(taskType);
+          break;
+        case SupportBundlePipelineInfoTask:
+          pipelineInfoTaskStatus = taskStatusMap.get(taskType);
+          break;
+        case SupportBundleRuntimeInfoTask:
+          runtimeInfoTaskStatus = taskStatusMap.get(taskType);
+          break;
+        case SupportBundlePipelineRunLogTask:
+          runtimeLogTaskStatus = taskStatusMap.get(taskType);
+          break;
+        case SupportBundleVMInfoTask:
+          vmInfoTaskStatus = taskStatusMap.get(taskType);
+          break;
+      }
+    }
+    return new SupportBundleTaskReportStatus(systemLogTaskStatus, pipelineInfoTaskStatus, runtimeInfoTaskStatus,
+                                             runtimeLogTaskStatus, vmInfoTaskStatus);
+  }
+
+  private static void zipFile(File fileToZip, String fileName, ZipOutputStream zipOut) throws IOException {
```

Comment:
There is a `BundleJarUtil.addToArchive` method doing the same. No need to write it again.